### PR TITLE
Fetch hci data and events until no more available

### PR DESCRIPTION
**Pull request by rugeGerritsen**
Fixes the situation where the hci driver could be waiting for a host signal even though there is events or data available

### CODE REVIEW

**Comment by joerchan** [comment]
@pdunaj Does this fix the issue you were seeing?

**Comment by pdunaj** [comment]
> @pdunaj Does this fix the issue you were seeing?

Unfortunately no. More information passed off-line.

File: subsys/bluetooth/controller/hci_driver.c [hunk 1]
```diff
@@ -203,54 +204,69 @@ static void event_packet_process(u8_t *hci_buf)
...
-			}
-			if (!errcode) {
-				data_packet_process(hci_buffer);
-			} else {
-				break;
-			}
-		}
+		received_evt = fetch_and_process_hci_evt(&hci_buffer[0]);
+
+		received_data = fetch_and_process_acl_data(&hci_buffer[0]);
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply subsys/bluetooth/controller/hci_driver.c]: On the other hand is the hci_buffer used anywhere beside inside of the fetch and process functions?

[the remark this answers is not in the document]
Comment (rugeGerritsen) [reply subsys/bluetooth/controller/hci_driver.c]: It is not

**Pull request merged**